get_google_ads_data.py

```
@@ -0,0 +1,156 @@
+
+      dict_location = {}
+
+      for batch in stream_location:
+        for row in batch.results:
+          dict_location[campaign.id] = row.campaign_criterion.proximity.address.city_name
+
+      df_campaign['country'] = df_campaign['campaign_id'].map(dict_location)
+
+      # get conversion data
+      query_conversion = """ 
+      SELECT 
+      conversion_action.name
+      FROM conversion_action
+      """
+
+      search_request.query = query_conversion
+      stream_conversion = ga_service.search_stream(search_request)
+
+      for batch in stream_conversion:
+        for row in batch.results:
+
+          conversion_name = row.conversion_action.name
+          metrics_conversions = row.metrics.all_conversions
+
+          pd_conversion = pd.Series(conversion_name)
+
+          if conversion_name == 'Добавление в корзину':
+            df_campaign.insert(7, "adds_to_cart", metrics_conversions, False)
+          if conversion_name == 'Покупка':
+            df_campaign.insert(7, "purchases", metrics_conversions, False)
+          if pd_conversion.str.startswith('Установки приложения').bool() == True:
+            df_campaign.insert(7, "ad_network_installs", metrics_conversions, False)
+
+      df = df_campaign.drop(columns = 'campaign_id')
+      col = df.columns.tolist()
+      col = col[ : 2] + col[ (len(col) - 3) : ] + col[ 2 : (len(col) - 3) ]
+      col = col[ : 8] + col[10:11] + col[8 : 9] + col[11 : 12] + col[9 : 10] + col[12 :]
+      df = df[col]
+
+      now_date = time.strftime("%d-%m-%Y_%H:%M")
+      file_csv = 'Report_google_ads/' + str(now_date) + '.csv'
+
+      if os.path.isfile(str(now_date) + '.csv') not in os.listdir('Report_google_ads'):
+        pd.DataFrame.to_csv(df, file_csv)
+        print(f'\nОтчет успешно выгружен в папку {file_csv}')
+      else:
+        print('\nОшибка при создании файла!')
+      
+  except:
+    print("\nОтчет с Google Ads не выгружен! Попробуйте еще раз!")
```

один bare except на абсолютно всю логику работы
тут может пойти что угодно не так - от проблем с авторизацией, до того, что просто ошибка в твоем коде.

при этом, без дебага никогда никто не узнает, а что же пошло не так.